File: week1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    // gets cents
    int cent;
    do
    {
        cent = get_int("Change owed: ");
    }
    while (cent < 0);
    return cent;
}

int calculate_quarters(int cents)
{
    // gets quarters
    int quarter = 0;
    while (cents >= 25)
    {
        cents -= 25;
        quarter++;
    }
    return quarter;
}

int calculate_dimes(int cents)
{
    // gets dimes
    int dime = 0;
    while (cents >= 10)
    {
        cents -= 10;
        dime++;
    }
    return dime;
}

int calculate_nickels(int cents)
{
    // gets nickels
    int nickel = 0;
    while (cents >= 5)
    {
        cents -= 5;
        nickel++;
    }
    return nickel;
}

int calculate_pennies(int cents)
{
    // gets pennies
    int penny = 0;
    while (cents >= 1)
    {
        cents -= 1;
        penny++;
    }
    return penny;
}

File: week6/sentimental-cash/cash.py
# importing cs50 library to take input
import cs50


# main function
def main():
    # gets user input change in float and rounds it to integer cents
    change = get_cents()
    change = round(change * 100)

    # calculates number of quarter coins
    quarters = calculate_quarters(change)
    change -= quarters * 25

    # calculates number of dime coins
    dimes = calculate_dimes(change)
    change -= dimes * 10

    # calculates number of nickel coins
    nickels = calculate_nickels(change)
    change -= nickels * 5

    # calculates number of penny coins
    penny = calculate_pennies(change)
    change -= penny

    # calculating total number of coins
    coins = quarters + dimes + nickels + penny
    print(coins)


# function to get user input for the amount of change
def get_cents():
    cents = -1
    while cents < 0:
        cents = cs50.get_float("Change owed: ")
    return cents


# function to calculate number of quarter coins for the given amount of change
def calculate_quarters(cent):
    quarters = 0
    while cent >= 25:
        cent -= 25
        quarters += 1
    return quarters


# function to calculate number of dime coins for the given amount of change
def calculate_dimes(cent):
    dimes = 0
    while cent >= 10:
        cent -= 10
        dimes += 1
    return dimes


# function to calculate number of nickel coins for the given amount of change
def calculate_nickels(cent):
    nickels = 0
    while cent >= 5:
        cent -= 5
        nickels += 1
    return nickels


# function to calculate number of penny coins for the given amount of change
def calculate_pennies(cent):
    penny = 0
    while cent >= 1:
        cent -= 1
        penny += 1
    return penny


main()
